src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProviderFields.cs

```diff
@@ -39,8 +39,20 @@ public ModelTypeProviderFields(InputModelType inputModel, TypeFactory typeFactor
 
             var visitedMembers = new HashSet<ISymbol>(SymbolEqualityComparer.Default);
 
-            foreach (var inputModelProperty in inputModel.Properties)
+            foreach (var property in inputModel.Properties)
             {
+                var inputModelProperty = property;
+                var name = BuilderHelpers.DisambiguateName(inputModel.Name.ToCleanName(), property.Name);
+                if (property.Name != name) // the previous name is ambiguous, need to update the name
+                {
+                    inputModelProperty = new InputModelProperty(name,
+                        property.SerializedName,
+                        property.Description,
+                        property.Type,
+                        property.IsRequired,
+                        property.IsReadOnly,
+                        property.IsDiscriminator);
+                }
```

Comment:
By reading the code below, it is unnecessary to create a new InputModelProperty here.
This new property is duplicating everything except for the name.
But code below are using the local variable `originalFieldName` for the name of the property.
Therefore we just need to assign the new name to `originalFieldName` and everything should work fine.

Reply:
But it will affected the serialize and deserialize. We only update the fieldName in the Model struct, but we need to use origin name for serialize/deserialize, that why we need to create the new InputModelProperty

Reply:
Thanks. `originalFieldName` is enough. I remove this code snippet.